### ISSUE TITLE
[20.04] Point character in Broker endpoint name prevents Broker to start (silently)

### ISSUE THREAD

**Issue by lorantino** [centreon/centreon#8771]
# BUG REPORT INFORMATION

### Prerequisites

> The opened issue, must be code related. GitHub is not meant for support. Feel free to check the CONTRIBUTING section for more details.

***Versions***

Central and poller in same version, this is from Central
```
rpm -qa | grep centreon |grep -v plugin
centreon-common-20.04.2-4.el7.centos.noarch
centreon-trap-20.04.2-4.el7.centos.noarch
centreon-widget-grid-map-20.04.0-5.el7.centos.noarch
centreon-license-manager-common-20.04.2-1.el7.centos.noarch
centreon-broker-20.04.4-2.el7.centos.x86_64
centreon-widget-host-monitoring-20.04.2-1.el7.centos.noarch
centreon-broker-cbmod-20.04.4-2.el7.centos.x86_64
centreon-connector-ssh-20.04.0-2.el7.centos.x86_64
centreon-auto-discovery-server-20.04.1-2.el7.centos.noarch
centreon-widget-servicegroup-monitoring-20.04.0-5.el7.centos.noarch
centreon-cwrapper-perl-0.1-1.el7.centos.x86_64
centreon-engine-daemon-20.04.1-1.el7.centos.x86_64
centreon-engine-extcommands-20.04.1-1.el7.centos.x86_64
centreon-poller-centreon-engine-20.04.2-4.el7.centos.noarch
centreon-clib-20.04.0-7.el7.centos.x86_64
centreon-widget-tactical-overview-20.04.0-5.el7.centos.noarch
centreon-pp-manager-20.04.1-1.el7.centos.noarch
centreon-perl-libs-20.04.2-4.el7.centos.noarch
centreon-engine-20.04.1-1.el7.centos.x86_64
centreon-web-20.04.2-4.el7.centos.noarch
centreon-base-config-centreon-engine-20.04.2-4.el7.centos.noarch
centreon-connector-20.04.0-2.el7.centos.x86_64
centreon-gorgone-20.04.2-1.el7.centos.noarch
centreon-broker-storage-20.04.4-2.el7.centos.x86_64
centreon-release-20.04-1.el7.centos.noarch
centreon-widget-httploader-20.04.0-5.el7.centos.noarch
centreon-widget-live-top10-cpu-usage-20.04.0-5.el7.centos.noarch
centreon-broker-core-20.04.4-2.el7.centos.x86_64
centreon-20.04.2-4.el7.centos.noarch
centreon-connector-perl-20.04.0-2.el7.centos.x86_64
centreon-gorgone-centreon-config-20.04.2-1.el7.centos.noarch
centreon-widget-live-top10-memory-usage-20.04.0-5.el7.centos.noarch
centreon-widget-service-monitoring-20.04.2-1.el7.centos.noarch
centreon-widget-engine-status-20.04.1-1.el7.centos.noarch
centreon-widget-hostgroup-monitoring-20.04.0-5.el7.centos.noarch
centreon-license-manager-20.04.2-1.el7.centos.noarch
centreon-widget-graph-monitoring-20.04.0-5.el7.centos.noarch
centreon-database-20.04.2-4.el7.centos.noarch
centreon-broker-cbd-20.04.4-2.el7.centos.x86_64
centreon-widget-global-health-20.04.0-5.el7.centos.noarch
```

***Operating System***

```
cat /etc/redhat-release
CentOS Linux release 7.8.2003 (Core)
```

***Browser used***

- [x] Google Chrome
- [ ] Firefox
- [ ] Internet Explorer IE11
- [ ] Safari

Version: 81.0.4044.138 (64bits)

***Additional environment details (AWS, VirtualBox, physical, etc.):***
VMware virtual machines (central=ova additional poller=centos7 iso)

### Description

This is a fresh install of Centreon 20.04.
The central-engine service on additional poller is running but do not connect to centreon-broker service on central server.
The gorgone-proxy service on central is well connected to gorgone service on poller.
See below to detailed logs.

### Steps to Reproduce

Once the additional poller is installed from official centos7 iso
1. I follow the official tutorial to configure the new poller (https://docs.centreon.com/current/en/monitoring/monitoring-servers/add-a-poller-to-configuration.html)
2. I use the wizard to connect it (/centreon/poller-wizard/1)
3. I choose new gorgone service to connect it (SSH method do not work too)
3. Once configured i copy paste the gorgone config from webui
4. Restart the service on poller and on central
5. Next I export the config to both servers and restart : Export configuration
6. Poller stay not running from central webui

### Describe the received result

Poller stay not running from central webui

### Describe the expected result

Established TCP flow between poller engine and central broker.
Poller in running state on webui.

### Logs

**PHP error logs**

Central
```
[04-Jun-2020 21:29:32 Europe/Paris] PHP Notice:  Undefined index: gorgone_api_username in /usr/share/centreon/src/Centreon/Domain/Gorgone/ConfigurationLoader.php on line 97
[04-Jun-2020 21:29:32 Europe/Paris] PHP Notice:  Undefined index: gorgone_api_password in /usr/share/centreon/src/Centreon/Domain/Gorgone/ConfigurationLoader.php on line 108
```

**centreon-engine logs (*if needed*)**

Poller
```
[1591299070] [1624] Centreon Engine 20.04.1 starting ... (PID=1624)
[1591299070] [1624] Local time is Thu Jun  4 21:31:10 2020
[1591299070] [1624] LOG VERSION: 2.0
[1591299070] [1624] Centreon Broker: log applier: applying 1 logging objects
[1591299070] [1624] Event broker module '/usr/lib64/nagios/cbmod.so' initialized successfully
[1591299070] [1624] Event broker module '/usr/lib64/centreon-engine/externalcmd.so' initialized successfully
[1591299070] [1624] Event loop start at Thu Jun  4 21:31:10 2020
[1591299070] [1624] Configuration loaded, main loop starting.
```

**centreon-broker logs (*if needed*)**

Central
```
Centreon Broker 20.04.4 log file closed
Centreon Broker 20.04.4 log file opened
[1591299077] error:   TCP peer 'localhost:5669'connection reset
```

**gorgoned logs**

Poller
```
2020-06-04 21:48:11 - INFO - [core] Initialize server keys
2020-06-04 21:48:11 - INFO - [core] Private key file '/var/lib/centreon-gorgone/.keys/rsakey.priv.pem' loaded
2020-06-04 21:48:11 - INFO - [core] Public key file '/var/lib/centreon-gorgone/.keys/rsakey.pub.pem' loaded
2020-06-04 21:48:11 - INFO - [core] Module 'action' is loading
2020-06-04 21:48:11 - INFO - [core] Module 'action' is loaded
2020-06-04 21:48:11 - INFO - [core] Module 'engine' is loading
2020-06-04 21:48:11 - INFO - [core] Module 'engine' is loaded
2020-06-04 21:48:11 - INFO - [core] Module 'dbcleaner' is loading
2020-06-04 21:48:11 - INFO - [core] Module 'dbcleaner' is loaded
2020-06-04 21:48:11 - INFO - [core] Gorgoned started
2020-06-04 21:48:11 - INFO - [core] PID 1677
2020-06-04 21:48:11 - INFO - [dbcleaner] Create module 'dbcleaner' process
2020-06-04 21:48:11 - INFO - [engine] Create module 'engine' process
2020-06-04 21:48:11 - INFO - [action] Create module 'action' process
2020-06-04 21:48:11 - INFO - [core] Server accepting clients
2020-06-04 21:48:26 - INFO - [core] Connection from proxy-1-10
```

Central
```
2020-06-04 21:48:26 - INFO - [core] Initialize server keys
2020-06-04 21:48:26 - INFO - [core] Private key file '/var/lib/centreon-gorgone/.keys/rsakey.priv.pem' loaded
2020-06-04 21:48:26 - INFO - [core] Public key file '/var/lib/centreon-gorgone/.keys/rsakey.pub.pem' loaded
2020-06-04 21:48:26 - INFO - [core] Module 'httpserver' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'httpserver' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'action' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'action' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'cron' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'cron' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'nodes' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'nodes' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'proxy' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'proxy' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'legacycmd' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'legacycmd' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'engine' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'engine' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'statistics' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'statistics' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'autodiscovery' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'autodiscovery' is loaded
2020-06-04 21:48:26 - INFO - [core] Module 'dbcleaner' is loading
2020-06-04 21:48:26 - INFO - [core] Module 'dbcleaner' is loaded
2020-06-04 21:48:26 - INFO - [core] Gorgoned started
2020-06-04 21:48:26 - INFO - [core] PID 3088
2020-06-04 21:48:26 - INFO - [nodes] Create module 'nodes' process
2020-06-04 21:48:26 - INFO - [dbcleaner] Create module 'dbcleaner' process
2020-06-04 21:48:26 - ERROR - [proxy] Cannot create child, need a core id
2020-06-04 21:48:26 - ERROR - [proxy] Cannot create child, need a core id
2020-06-04 21:48:26 - ERROR - [proxy] Cannot create child, need a core id
2020-06-04 21:48:26 - ERROR - [proxy] Cannot create child, need a core id
2020-06-04 21:48:26 - ERROR - [proxy] Cannot create child, need a core id
2020-06-04 21:48:26 - INFO - [autodiscovery] Create module 'autodiscovery' process
2020-06-04 21:48:26 - INFO - [cron] Create module 'cron' process
2020-06-04 21:48:26 - INFO - [engine] Create module 'engine' process
2020-06-04 21:48:26 - INFO - [statistics] Create module 'statistics' process
2020-06-04 21:48:26 - INFO - [action] Create module 'action' process
2020-06-04 21:48:26 - INFO - [httpserver] Create module 'httpserver' process
2020-06-04 21:48:26 - INFO - [legacycmd] Create module 'legacycmd' process
2020-06-04 21:48:26 - INFO - [core] Server accepting clients
2020-06-04 21:48:26 - INFO - [cron] Adding cron definition 'broker_stats'
2020-06-04 21:48:26 - INFO - [cron] Adding cron definition 'engine_stats'
2020-06-04 21:48:26 - INFO - [proxy] Create module 'proxy' child process for pool id '1'
2020-06-04 21:48:26 - INFO - [proxy] Create module 'proxy' child process for pool id '2'
2020-06-04 21:48:26 - INFO - [proxy] Create module 'proxy' child process for pool id '3'
2020-06-04 21:48:26 - INFO - [proxy] Create module 'proxy' child process for pool id '4'
2020-06-04 21:48:26 - INFO - [proxy] Create module 'proxy' child process for pool id '5'
2020-06-04 21:48:26 - INFO - [core] Setcoreid changed 1
2020-06-04 21:48:26 - INFO - [proxy] Node '10' is registered
2020-06-04 21:48:26 - INFO - [zmqclient] Client connected successfully to 'tcp://10.10.4.100:5556'
2020-06-04 21:48:26 - INFO - [proxy] Pong received from '10'
2020-06-04 21:48:46 - INFO - [proxy] Send pings
2020-06-04 21:48:46 - INFO - [proxy] Pong received from '10'
```

**Comment by lorantino** [comment]
I tried to install the 19.10 version with same configuration and found the issue.
The broker log contain the related issue without turn on debug on webui :
state applier: broker_name is not  valid: allowed characters are ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 -_

Unfortunately for the 20.04 version there is no related output on broker log and no check for characters input on the poller name field.

Laurent

**Comment by cgagnaire** [comment]
Hi @lorantino,
Indeed there is a set of caracters to be used in the endpoint name.
It has been identified and we will provide with fixes on both UI and Broker.

Refs MON-5644 MON-5656

**Comment by omercier** [comment]
Hi @lorantino,
Sorry for the long wait. This issue has been fixed since release 20.04.13.
https://docs.centreon.com/20.04/en/releases/centreon-core.html#200413-2
